pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -355,13 +356,37 @@ public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingCon
     indexLoadingConfig.setInstanceTierConfigs(_tableDataManagerConfig.getInstanceTierConfigs());
     File indexDir = getSegmentDataDir(segmentName, segmentTier, indexLoadingConfig.getTableConfig());
     try {
+      boolean shouldDownload = forceDownload || !hasSameCRC(zkMetadata, localMetadata);
+
+      if (!shouldDownload && (zkMetadata.getTier().equals(segmentTier))) {
```

Comment:
@klsince Owing to the complexities involving TierBasedSegmentDirectoryLoader::load (it can move an existing non backup directory to new data dir if not used with a backup dir, etc), I think it might make sense to use this exclusively for cases where no tier change is involved. LMK what you think?